improver/utilities/mathematical_operations.py

```diff
@@ -638,14 +651,306 @@ def process(
+        # mean cube.
+        # The verification of the standard deviation cube's bounds suitably matching the
+        # anomaly cube's is covered implicitly due to the above code chunk.
+        if not self.ignore_temporal_mismatch:
+            if anomaly_cube.coord("reference_epoch") is not None:
+                if np.any(
+                    anomaly_cube.coord("reference_epoch").points
+                    != mean_cube.coord("time").points
+                ) or not np.all(
+                    anomaly_cube.coord("reference_epoch").bounds
+                    == mean_cube.coord("time").bounds
+                ):
+                    errors.append(
+                        "The reference epoch coordinate of the anomaly cube must "
+                        "match the time coordinate of the mean cube. I.e. The same mean"
+                        "cube that was used to calculate the anomaly."
+                        "The following values were found: "
+                        f"anomaly cube reference epoch: {anomaly_cube.coord('reference_epoch').points},"
+                        f"mean cube time: {mean_cube.coord('time').points},"
+                        f"anomaly cube reference epoch bounds: {anomaly_cube.coord('reference_epoch').bounds},"
+                        f"mean cube time bounds: {mean_cube.coord('time').bounds}"
+                    )
+
+        if errors:
+            raise ValueError("\n".join(errors))
+
+    @staticmethod
+    def calculate_forecast_value(
+        anomaly_cube: Cube, mean_cube: Cube, std_cube: Optional[Cube] = None
+    ) -> ndarray:
+        """Calculate forecast value(s) from the input cubes."""
+        if std_cube:
+            forecast_value = (anomaly_cube.data * std_cube.data) + mean_cube.data
+        else:
+            forecast_value = anomaly_cube.data + mean_cube.data
+
+        return forecast_value
+
+    @staticmethod
+    def _update_cube_name_and_units(
+        output_cube: Cube, mean_cube: Cube, standardized_anomaly: bool = False
+    ) -> None:
+        """This method removes the portion of the name of the output cube
+        cube that indicates that it is an anomaly and converts units to be dimensional
+        (i.e. not '1') if a standardized anomaly is used. The cube is modified in place.
+
+        Args:
+            output_cube:
+                The cube to be updated.
+            standardized_anomaly:
+                Flag indicating if the forecast values are calculated using
+                standardized anomalies. If True, a "_standardized_anomaly" suffix is
+                removed from the name. If False, an "_anomaly" suffix is removed from
+                the name.
+        """
+        suffix = "_standardized_anomaly" if standardized_anomaly else "_anomaly"
+
+        try:
+            output_cube.standard_name = output_cube.standard_name.replace(suffix, "")
+        except ValueError:
+            output_cube.long_name = output_cube.long_name.replace(suffix, "")
```

Comment:
```suggestion
    output_cube.standard_name = output_cube.standard_name.replace(suffix, "")
    if output_cube.long_name:
        output_cube.long_name = output_cube.long_name.replace(suffix, "")
```
Attempting to replace a sub-string which isn't in the original string doesn't result in an error. We can use this method instead.

Reply:
Good to know. Thank you.